src/OrbitGl/TimelineUiTest.cpp

```diff
@@ -0,0 +1,95 @@
+    int num_boxes = mock_batcher_.GetNumBoxes();
+    int num_major_ticks = mock_batcher_.GetNumLinesByColor(kMajorTickColor);
+    int num_minor_ticks = mock_batcher_.GetNumLinesByColor(kMinorTickColor);
+
+    // Lines should be only major and minor ticks.
+    EXPECT_EQ(mock_batcher_.GetNumLines(), num_major_ticks + num_minor_ticks);
+
+    // Major ticks should always be between 2 and 10, given scale set. In extremely small screens
+    // (Width < 160), we don't force a minimum number of major ticks.
+    if (!is_small_screen) EXPECT_GE(num_major_ticks, 2);
+    EXPECT_LE(num_major_ticks, 10);
+
+    // Depending on the scale, there should be 1 minor ticks or 4 between each major tick, which
+    // means that (calling MT to the number of major ticks, and mt to the number of minor ticks)
+    // mt is between (MT-1, MT+1) or mt is between (4x(MT-1), 4x(MT+1)).
+    EXPECT_GE(num_minor_ticks, num_major_ticks - 1);
+    EXPECT_LE(num_minor_ticks, 4 * (num_major_ticks + 1));
+    EXPECT_TRUE(num_minor_ticks <= num_major_ticks + 1 ||
+                num_minor_ticks >= 4 * (num_major_ticks - 1));
+
+    // Labels should all have the same number of numbers, start at the same vertical position and
+    // they will appear at the right of each major tick.
+    EXPECT_TRUE(mock_text_renderer_.HasAddTextsSameLength());
+    EXPECT_TRUE(mock_text_renderer_.AreAddTextsAlignedVertically());
+    EXPECT_GE(num_labels, num_major_ticks - 1);
+    EXPECT_LE(num_labels, num_major_ticks + 1);
+
+    // Boxes: One box per each label + Background box + margin box.
+    EXPECT_EQ(num_boxes, num_labels + 2);
+
+    // Everything should be between kZValueTimeBar and kZValueTimeBarMouseLabel.
+    EXPECT_TRUE(mock_text_renderer_.IsTextBetweenZLayers(GlCanvas::kZValueTimeBar,
+                                                         GlCanvas::kZValueTimeBarMouseLabel));
+    EXPECT_TRUE(mock_batcher_.IsEverythingBetweenZLayers(GlCanvas::kZValueTimeBar,
+                                                         GlCanvas::kZValueTimeBarMouseLabel));
+  }
+
+ private:
+  MockBatcher mock_batcher_;
+  MockTextRenderer mock_text_renderer_;
+  PrimitiveAssembler primitive_assembler_;
+  TimeGraphLayout layout_;
+  Viewport viewport_;
+  MockTimelineInfo mock_timeline_info;
+};
+
+TEST(TimelineUI, UpdatePrimitivesBigScreen) {
+  // TODO(b/218311326): In some cases, small screens and even medium screens (Width < 700) failed in
+  // the condition between #labels and #major_ticks, since we are drawing major_ticks anyways when
+  // labels intersect.
+  const int kBigScreenWidth = 2000;
+  TimelineUiTest timeline_ui_test(kBigScreenWidth);
+  timeline_ui_test.TestUpdatePrimitives(0, 100);
+  for (int i = 0; i < 100; i++) {
+    uint64_t timestamp_1 = rand();
+    uint64_t timestamp_2 = rand();
+    timeline_ui_test.TestUpdatePrimitives(std::min(timestamp_1, timestamp_2),
+                                          std::max(timestamp_1, timestamp_2));
+  }
```

Comment:
I am not sure how I feel about this style of unit test. I understand that this seems a good exhaustive way of testing this, but there are also (potential) downsides of introducing randomness. 
* Random values can be the source of flakyness
* Unit test coverage might be changing 

Also, there is absl random, which might be more appropriate than `rand()`. Maybe a hardcoded seed value is an option to make this deterministic, but maybe this has other downsides. 

Also, what about edge cases like `timestamp1 == timestamp2` ? This could potentially happen here, but has a very low probability.

Reply:
> Random values can be the source of flakyness
> Unit test coverage might be changing

No, they should always be the same, I'm not choosing a seed based on time.


> Also, there is absl random, which might be more appropriate than rand()
I wasn't aware of this. Is this used somewhere else?

> Also, what about edge cases like timestamp1 == timestamp2 ? 
Good point. We are not allowing this case, the maximum zooming level is 100 nanoseconds. But it shouldn't fail anyways.